redash/handlers/dashboards.py

```diff
@@ -149,9 +149,12 @@ def get(self, dashboard_slug=None):
         :>json string widget.created_at: ISO format timestamp for widget creation
         :>json string widget.updated_at: ISO format timestamp for last widget modification
         """
-        dashboard = get_object_or_404(
-            models.Dashboard.get_by_slug_and_org, dashboard_slug, self.current_org
+        fn = (
+            models.Dashboard.get_by_slug_and_org
+            if request.args.get("legacy") is not None
+            else models.Dashboard.get_by_id_and_org
         )
```

Comment:
Ternary operators aren't that readable in Python. Might be better to write this as:

```python
if request.args.get("legacy") is not None:
    fn = models.Dashboard.get_by_slug_and_org
else:
    fn = models.get_by_id_and_org
```

(`fn` will be available in the outer context)

Reply:
Done 👍